src/main/java/com/cloudant/client/org/lightcouch/CouchDatabaseBase.java

```diff
@@ -267,18 +269,28 @@ public Response remove(String id, String rev) {
      */
     public List<Response> bulk(List<?> objects, boolean allOrNothing) {
         assertNotEmpty(objects, "objects");
-        InputStream response = null;
+        InputStream responseStream = null;
+        HttpConnection connection;
         try {
             final String allOrNothingVal = allOrNothing ? "\"all_or_nothing\": true, " : "";
             final URI uri = buildUri(getDBUri()).path("_bulk_docs").build();
             final String json = String.format("{%s%s%s}", allOrNothingVal, "\"docs\": ", getGson
                     ().toJson(objects));
-            response = couchDbClient.post(uri, json);
-            return getResponseList(response, getGson(), Response.class,
+            connection = couchDbClient.post(uri, json, HttpConnection.class);
+            responseStream = connection.responseAsInputStream();
+            List<Response> bulkResponses = getResponseList(responseStream, getGson(),
                     new TypeToken<List<Response>>() {
                     }.getType());
+            for(Response response : bulkResponses) {
+                response.setStatusCode(connection.getConnection().getResponseCode());
+                response.setReason(connection.getConnection().getResponseMessage());
```

Comment:
I think this is the best that can be done for now with the current `List<Response>`, but at some point in the future it will probably make more sense to have a different return type that could hold the HTTP status code, and had a getter to unpack the list of individual responses.